server/game/GameActions/AttachAction.ts

```diff
@@ -28,6 +35,8 @@ export class AttachAction extends CardGameAction {
             return false;
         } else if(!properties.attachment || properties.attachment.anotherUniqueInPlay(context.player) || !properties.attachment.canAttach(card, context)) {
             return false;
+        } else if(properties.takeControl && properties.attachment.controller === context.player) {
```

Comment:
I'm not sure this is a great idea. An effect which moves an attachment and takes control of it is still a game state change if it cannot take control (e.g. Karada District)

Reply:
As discussed, there are currently no effects that independently take control and/or move an attachment.  So we need this still to check for Karada/CiF.